src/dotnet/CommandFactory/CommandResolution/LocalToolsCommandResolver.cs

```diff
@@ -32,7 +32,7 @@ internal class LocalToolsCommandResolver : ICommandResolver
             _fileSystem = fileSystem ?? new FileSystemWrapper();
         }
 
-        public CommandSpec Resolve(CommandResolverArguments arguments)
+        public CommandSpec ResolveStrict(CommandResolverArguments arguments)
```

Comment:
Is this only for test purposes now?

Reply:
in "dotnet tool run"

you can only call `dotnet-mytool` with `dotnet tool run dotnet-mytool` 
but you could do `dotnet mytool` and `dotnet dotnet-mytool`

Reply:
oop, i did not change dotnet tool run to use resolvestrcit. changing